test/addons-napi/test_fatal/test_fatal.c

```diff
@@ -0,0 +1,18 @@
+#include <node_api.h>
+#include "../common.h"
+
+napi_value Test(napi_env env, napi_callback_info info) {
+  napi_fatal_error("test_fatal::Test", "fatal message");
+  return NULL;
+}
+
+void Init(napi_env env, napi_value exports, napi_value module, void* priv) {
+  napi_property_descriptor properties[] = {
+    DECLARE_NAPI_PROPERTY("Test", Test),
+  };
+
+  NAPI_CALL_RETURN_VOID(env, napi_define_properties(
+    env, exports, sizeof(properties) / sizeof(*properties), properties));
```

Comment:
Tangential: macros like NAPI_ASSERT_BASE and NAPI_CALL_BASE should have their bodies wrapped in `do { } while (0)` in order for them to be safe to use in if/else statements.  Right now they can create bugs that will be hard to figure out.  Something for another PR.